### DISCUSSION TITLE
Is it possible to supply the ID when adding a new row?

### DISCUSSION THREAD

**Discussion by marbemac** [tinyplex/tinybase#103] [Q&A]
For use cases where IDs are known in advance, or generated / managed outside of tinybase?

My specific use case atm is a daily journaling app, where each journal entry is uniquely identified by the day (`2023-10-27`).

For the moment I'm using the `useSetCellCallback` to create new rows, setting a dummy property on the entry. E.g. `const addEntry = useSetCellCallback('entries', dayId, 'placeholderProp', () => '@TODO', []);`

**Comment by marbemac** [comment]
@jamesgpearce, is the recommended way to add a row when you know the ID in advance, any downsides vs the built-in addRow?

**Reply by jamesgpearce** [reply]
Could you use [setRow](https://tinybase.org/api/store/interfaces/store/store/methods/setter/setrow/) or [useSetRowCallback](https://tinybase.org/api/ui-react/functions/store-hooks/usesetrowcallback/)?
